src/sys/socket/addr.rs

```diff
@@ -730,13 +814,31 @@ impl SockAddr {
                 #[cfg(any(target_os = "ios", target_os = "macos"))]
                 Some(AddressFamily::System) => Some(SockAddr::SysControl(
                     SysControlAddr(*(addr as *const sys_control::sockaddr_ctl)))),
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                Some(AddressFamily::Packet) => Some(SockAddr::Ether(
+                    EtherAddr(*(addr as *const libc::sockaddr_ll)))),
+                #[cfg(any(target_os = "dragonfly",
+                          target_os = "freebsd",
+                          target_os = "ios",
+                          target_os = "macos",
+                          target_os = "netbsd",
+                          target_os = "openbsd"))]
+                Some(AddressFamily::Link) => {
+                    let ether_addr = EtherAddr(*(addr as *const libc::sockaddr_dl));
+                    if ether_addr.is_empty() {
+                        None
+                    } else {
+                        Some(SockAddr::Ether(ether_addr))
+                    }
+                },
                 // Other address families are currently not supported and simply yield a None
                 // entry instead of a proper conversion to a `SockAddr`.
                 Some(_) | None => None,
             }
         }
     }
 
+    /// Conversion SockAddr to a libc's sockaddr.
```

Comment:
"Conversion from `nix`'s `SockAddr` type to the underlying `libc` `sockaddr` type.

This is useful for interfacing with other `libc` functions that don't yet have `nix` wrappers.

Returns a reference to the underlying data type (as a `sockaddr` reference) along with the size of the actual data type. `sockaddr` is commonly used as a proxy for a superclass as C doesn't support inheritance, so many functions that take a `sockaddr *` need to take the size of the underlying type as well and then internally cast it back.

Reply:
okay, and how do u think rename `EtherAddr` to `LinkAddr` or `HwAddr` ?

Reply:
I think `LinkAddr` would work nicely.